In [87]:
import pandas as pd 
import numpy as np


### Setting up each file as a DataFrame and checking that they can be read. 

In [88]:
bech_df = pd.read_csv("../Data/Bechdel_IMDB_Merge0524.csv")
cin_df = pd.read_csv("../Data/cinatomy_movie_profiles.csv")
osc_df = pd.read_csv("../Data/oscars_df.csv")
horror_df = pd.read_csv("../Data/horror_movies.csv")


In [89]:
bech_df.head()

,title,year,imdbid,id,bechdelRating,imdbAverageRating,numVotes,runtimeMinutes,genre1,genre2,genre3
0,Miss Jerry,1894,9.0,9779,0,5.4,212.0,45,Romance,NaN,NaN
1,"Story of the Kelly Gang, The",1906,574.0,1349,1,6.0,903.0,70,Action,Adventure,Biography
2,Cleopatra,1912,2101.0,2003,2,5.1,622.0,100,Drama,History,NaN
3,A Florida Enchantment,1914,3973.0,4457,2,5.8,300.0,63,Comedy,NaN,NaN
4,"Birth of a Nation, The",1915,4972.0,1258,2,6.1,26403.0,195,Drama,History,War


In [90]:
cin_df.head()

,movie_title,pacing_efficiency,originality,family_friendly,plot_quality_rating,plot_quality_reasons_for_liking,plot_quality_reasons_for_disliking,dialogues,end_feeling_emotional_adjective,end_feeling_ending_rating,...,sound_bgm_quality,sound_song_tracks_quality,performance_of_actors,connected,references_contained,cognitive_requirement,technical_knowledge_required,photosensitivity_warnings,animal_harm,trailer_or_spoiler
0,The Shawshank Redemption,4,4,2,5,"['unpredictable', 'plot_twists', 'strong_pacin...",[],5,happy,5,...,5,5,5,5,1,3,1,1,1,1
1,The Godfather,4,5,1,5,"['strong_pacing', 'tight_structure']",[],5,bittersweet,5,...,5,4,5,3,4,4,1,1,4,3
2,The Dark Knight,5,5,1,5,"['unpredictable', 'plot_twists', 'strong_pacin...",[],5,bittersweet,5,...,5,4,5,5,4,4,1,3,1,5
3,The Godfather: Part II,4,5,1,5,"['unpredictable', 'plot_twists', 'strong_pacin...",[],5,bittersweet,5,...,5,4,5,5,5,4,1,1,1,5
4,12 Angry Men,5,5,3,5,"['unpredictable', 'plot_twists', 'strong_pacin...",[],5,energized,5,...,3,1,5,1,1,4,1,1,1,3


In [91]:
osc_df.head()

,Unnamed: 0,Film,Oscar Year,Film Studio/Producer(s),Award,Year of Release,Movie Time,Movie Genre,IMDB Rating,IMDB Votes,...,Tomatometer Status,Tomatometer Rating,Tomatometer Count,Audience Status,Audience Rating,Audience Count,Tomatometer Top Critics Count,Tomatometer Fresh Critics Count,Tomatometer Rotten Critics Count,Film ID
0,0,Wings,1927/28,Famous Players-Lasky,Winner,1927,144,"Drama,Romance,War",7.5,"12,221",...,Certified-Fresh,93.0,46.0,Upright,78.0,3530.0,9.0,43.0,3.0,2becf7d5-a3de-46ab-ae45-abdd6b588067
1,1,7th Heaven,1927/28,Fox,Nominee,1927,110,"Drama,Romance",7.7,"3,439",...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,19ed3295-a878-4fd2-8e60-5cd7b5f93dad
2,2,The Racket,1927/28,The Caddo Company,Nominee,1928,84,"Crime,Drama,Film-Noir",6.7,"1,257",...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,3111c2d8-0908-4093-8ff3-99c89f2f2f08
3,3,The Broadway Melody,1928/29,Metro-Goldwyn-Mayer,Winner,1929,100,"Drama,Musical,Romance",5.7,"6,890",...,Rotten,33.0,24.0,Spilled,21.0,1813.0,7.0,8.0,16.0,de063f3f-2d35-4e1c-8636-6eb4c16bd236
4,4,Alibi,1928/29,Feature Productions,Nominee,1929,91,"Action,Crime,Romance",5.8,765,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,609887c2-877c-43a4-b88c-e40e31096a98


In [92]:
horror_df.head()

,id,original_title,title,original_language,overview,tagline,release_date,poster_path,popularity,vote_count,vote_average,budget,revenue,runtime,status,adult,backdrop_path,genre_names,collection,collection_name
0,760161,Orphan: First Kill,Orphan: First Kill,en,After escaping from an Estonian psychiatric fa...,There's always been something wrong with Esther.,2022-07-27,/pHkKbIRoCe7zIFvqan9LFSaQAde.jpg,5088.584,902,6.9,0,9572765,99,Released,False,/5GA3vV1aWWHTSDO5eno8V5zDo8r.jpg,"Horror, Thriller",760193.0,Orphan Collection
1,760741,Beast,Beast,en,A recently widowed man and his two teenage dau...,Fight for family.,2022-08-11,/xIGr7UHsKf0URWmyyd5qFMAq4d8.jpg,2172.338,584,7.1,0,56000000,93,Released,False,/2k9tBql5GYH328Krj66tDT9LtFZ.jpg,"Adventure, Drama, Horror",NaN,NaN
2,882598,Smile,Smile,en,"After witnessing a bizarre, traumatic incident...","Once you see it, it’s too late.",2022-09-23,/hiaeZKzwsk4y4atFhmncO5KRxeT.jpg,1863.628,114,6.8,17000000,45000000,115,Released,False,/mVNPfpydornVe4H4UCIk7WevWjf.jpg,"Horror, Mystery, Thriller",NaN,NaN
3,756999,The Black Phone,The Black Phone,en,"Finney Blake, a shy but clever 13-year-old boy...",Never talk to strangers.,2022-06-22,/lr11mCT85T1JanlgjMuhs9nMht4.jpg,1071.398,2736,7.9,18800000,161000000,103,Released,False,/AfvIjhDu9p64jKcmohS4hsPG95Q.jpg,"Horror, Thriller",NaN,NaN
4,772450,Presencias,Presences,es,A man who loses his wife and goes to seclude h...,NaN,2022-09-07,/dgDT3uol3mdvwEg0jt1ble3l9hw.jpg,1020.995,83,7.0,0,0,0,Released,False,/ojfzhdwRemcDt1I6pao6vVLw9AA.jpg,Horror,NaN,NaN


### Checking each dataset's info, shape, and if there are any duplicate rows.

In [93]:
bech_df.info()
bech_df.shape
bech_df.duplicated().value_counts()

<class 'pandas.DataFrame'>
RangeIndex: 9718 entries, 0 to 9717
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   title              9718 non-null   str    
 1   year               9718 non-null   int64  
 2   imdbid             9718 non-null   float64
 3   id                 9718 non-null   int64  
 4   bechdelRating      9718 non-null   int64  
 5   imdbAverageRating  9718 non-null   float64
 6   numVotes           9718 non-null   float64
 7   runtimeMinutes     9718 non-null   str    
 8   genre1             9718 non-null   str    
 9   genre2             8446 non-null   str    
 10  genre3             5807 non-null   str    
dtypes: float64(3), int64(3), str(5)
memory usage: 835.3 KB


False    9718
Name: count, dtype: int64

In [94]:
cin_df.info()
cin_df.shape
cin_df.duplicated().value_counts()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 27 columns):
 #   Column                              Non-Null Count  Dtype
---  ------                              --------------  -----
 0   movie_title                         1000 non-null   str  
 1   pacing_efficiency                   1000 non-null   int64
 2   originality                         1000 non-null   int64
 3   family_friendly                     1000 non-null   int64
 4   plot_quality_rating                 1000 non-null   int64
 5   plot_quality_reasons_for_liking     1000 non-null   str  
 6   plot_quality_reasons_for_disliking  1000 non-null   str  
 7   dialogues                           1000 non-null   int64
 8   end_feeling_emotional_adjective     1000 non-null   str  
 9   end_feeling_ending_rating           1000 non-null   int64
 10  end_feeling_structural_adjective    1000 non-null   str  
 11  immersive                           1000 non-null   int64
 12  impactful         

False    999
True       1
Name: count, dtype: int64

In [95]:
osc_df.info()
osc_df.shape
osc_df.duplicated().value_counts()


<class 'pandas.DataFrame'>
RangeIndex: 571 entries, 0 to 570
Data columns (total 30 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Unnamed: 0                        571 non-null    int64  
 1   Film                              571 non-null    str    
 2   Oscar Year                        571 non-null    str    
 3   Film Studio/Producer(s)           571 non-null    str    
 4   Award                             571 non-null    str    
 5   Year of Release                   571 non-null    int64  
 6   Movie Time                        571 non-null    int64  
 7   Movie Genre                       571 non-null    str    
 8   IMDB Rating                       571 non-null    float64
 9   IMDB Votes                        571 non-null    str    
 10  Movie Info                        438 non-null    str    
 11  Genres                            439 non-null    str    
 12  Critic Consensus   

False    571
Name: count, dtype: int64

In [96]:
horror_df.info()
horror_df.shape
horror_df.duplicated().value_counts()

<class 'pandas.DataFrame'>
RangeIndex: 32540 entries, 0 to 32539
Data columns (total 20 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 32540 non-null  int64  
 1   original_title     32540 non-null  str    
 2   title              32540 non-null  str    
 3   original_language  32540 non-null  str    
 4   overview           31254 non-null  str    
 5   tagline            12706 non-null  str    
 6   release_date       32540 non-null  str    
 7   poster_path        28066 non-null  str    
 8   popularity         32540 non-null  float64
 9   vote_count         32540 non-null  int64  
 10  vote_average       32540 non-null  float64
 11  budget             32540 non-null  int64  
 12  revenue            32540 non-null  int64  
 13  runtime            32540 non-null  int64  
 14  status             32540 non-null  str    
 15  adult              32540 non-null  bool   
 16  backdrop_path      13545 non-null

False    32540
Name: count, dtype: int64

Dropping duplicates 

In [97]:
cin_df.drop_duplicates(keep='first', inplace=True)
cin_df.duplicated().value_counts()

False    999
Name: count, dtype: int64

#### Checking for nulls, and determining if the columns need adjusting.

In [98]:
bech_df.isnull().sum()

title                   0
year                    0
imdbid                  0
id                      0
bechdelRating           0
imdbAverageRating       0
numVotes                0
runtimeMinutes          0
genre1                  0
genre2               1272
genre3               3911
dtype: int64

Only nulls in the Bechdel Test Dataframe are from subgenres(columns: genre2, genre3)
Most likely needs no adjusting. 

In [99]:
cin_df.isnull().sum()

movie_title                           0
pacing_efficiency                     0
originality                           0
family_friendly                       0
plot_quality_rating                   0
plot_quality_reasons_for_liking       0
plot_quality_reasons_for_disliking    0
dialogues                             0
end_feeling_emotional_adjective       0
end_feeling_ending_rating             0
end_feeling_structural_adjective      0
immersive                             0
impactful                             0
overall_feeling                       0
visuals                               0
visual_effects                        0
sound_audio_balance_issue             0
sound_bgm_quality                     0
sound_song_tracks_quality             0
performance_of_actors                 0
connected                             0
references_contained                  0
cognitive_requirement                 0
technical_knowledge_required          0
photosensitivity_warnings             0


No nulls found in Cinatomy DataFrame

In [100]:
osc_df.isnull().sum()

Unnamed: 0                            0
Film                                  0
Oscar Year                            0
Film Studio/Producer(s)               0
Award                                 0
Year of Release                       0
Movie Time                            0
Movie Genre                           0
IMDB Rating                           0
IMDB Votes                            0
Movie Info                          133
Genres                              132
Critic Consensus                    232
Content Rating                      132
Directors                           134
Authors                             134
Actors                              132
Original Release Date               132
Streaming Release Date              133
Production Company                  132
Tomatometer Status                  132
Tomatometer Rating                  132
Tomatometer Count                   132
Audience Status                     135
Audience Rating                     132


Presumably, 132 rows are missing information in the following columns:
- movie info*
- genres*
- critic consensus
- content rating
- directors
- authors
- actors
- original release date
- streaming release date
- production company
- ... and all columns involving Rotten Tomatoes

*( ***asterick*** where null in column might be a concern moving forward)*


In [101]:
print("Most Recent Release Date:")
print(osc_df['Original Release Date'].max())
print("Earliest Release Date:")
print(osc_df['Original Release Date'].min())


Most Recent Release Date:
2019-11-06
Earliest Release Date:
1927-08-12


The cause of missing values might be the earlier movies. 
***Look more into this in later cleaning.***

In [102]:
horror_df.isnull().sum()

id                       0
original_title           0
title                    0
original_language        0
overview              1286
tagline              19834
release_date             0
poster_path           4474
popularity               0
vote_count               0
vote_average             0
budget                   0
revenue                  0
runtime                  0
status                   0
adult                    0
backdrop_path        18995
genre_names              0
collection           30234
collection_name      30234
dtype: int64

Nulls in collection_name are fine as is, they are there to let us know that that particular movie is not part of a larger collection.

Nulls in tagline are also fine, not all movies have a tagline, especially the older movies in this dataframe. 
Might ultimately drop this column as it's not incredibly useful. 
Though it could be interesting to see why there is a lack of tagline? 

Columns to drop(due to nulls or repetition): 
- poster_path 
- backdrop_path
- collection (unnecessary since we have collection_name)
- adult(none of these movies are pornography, so all rows have same value)
I don't plan to use these columns and they have a significt number of nulls.
